src/front/glsl/functions.rs

```diff
@@ -535,6 +560,26 @@ impl Parser {
             TypeInner::Vector { size, kind, width } => {
                 return self.vector_constructor(ctx, body, ty, size, kind, width, &args, meta)
             }
+            TypeInner::Array { base, .. } => {
+                for (mut arg, meta) in args.iter().copied() {
+                    let scalar_components = scalar_components(&self.module.types[base].inner);
```

Comment:
would it be easier to do this via
```rust
impl crate::TypeInner {
  fn scalar_components(&self) -> ... {... }
}
```

Reply:
It would be almost the same thing, I have no preference so if you prefer the method call instead of a function call that's fine by me.